src/lib/ast/and.ts

```diff
@@ -0,0 +1,51 @@
+import { Skill } from "../types";
+import { SkillExpression } from "./skillExpression";
+import { createJson } from "./jsonHandler";
+import { Variable } from "./variable";
+import { SkillsRelations } from "./skillsRelation";
+
+// And skill expression
+export class And extends SkillExpression {
+
+	constructor(private terms: SkillExpression[]) {
+		super();
+		this.type = "And";
+
+	}
+
+	evaluate(learnedSkills: ReadonlyArray<string>, skillsRelations: SkillsRelations, without?: Variable[]): boolean {
+		let filterTerms:SkillExpression[] = [];
+		if (without) {
+			filterTerms = this.filterSkillsByWithout(skillsRelations, without);
+		} else {
+			filterTerms = this.terms;
+		}
+		return filterTerms.every(value => value.evaluate(learnedSkills, skillsRelations, without));
+	}
+
+	extractSkills(): Skill[] {
+		let skillList: Skill[] = [];
+		this.terms.forEach(expression => {
+			skillList.push(...expression.extractSkills());
+		});
+
+        return skillList;
+	}
+
+	toJson(): string {
+		return createJson(And.name, this.terms);
+	}
+
+	filterSkillsByWithout(skillsRelations: SkillsRelations, without?: Variable[]): SkillExpression[] {

```

Comment:
I'm concerned that this function won't work recursively. That happens in the following situation:
`OR(AND(A, B, C), AND(D, E, F)), WITHOUT (A, E)`
Will be there an exception for `A` and `E` when I evaluate the `OR`-expression?

Reply:
The recursion is achieved through the evaluate function, which passes the "without" list into the nested expressions for evaluation. 
Note that in this example, the "Without" list is ignored since the skills are explicitly listed. However, let's assume that is not the case.

<img width="498" alt="NestedSkillExpressionWithout" src="https://github.com/e-Learning-by-SSE/nm-skill-lib/assets/75734718/37f561ec-130c-47e1-9ff0-b9c3f07e571a">

Reply:
Looks good (convincing) 👍